src/ControlSystem/ControlErrors/Translation.hpp

```diff
@@ -109,30 +111,54 @@ struct Translation : tt::ConformsTo<protocols::ControlError> {
                                          grid_position_of_A_tnsr[2]}};
     const DataVector& current_position_of_A = get<center_A>(measurements);
 
-    const DataVector rotation_error =
-        rotation_control_error_(tuner, cache, time, "Rotation", measurements);
-    // Use A because it's on the positive x-axis, however, B would work as well.
-    // Just so long as we are consistent.
-    const DataVector rotation_error_cross_grid_pos_A =
-        cross(rotation_error, grid_position_of_A);
-
-    const double expansion_error = expansion_control_error_(
-        tuner, cache, time, "Expansion", measurements)[0];
-
-    // From eq. 42 in 1304.3067
-    const quat middle_expression = datavector_to_quaternion(
-        current_position_of_A -
-        (1.0 + expansion_error / expansion_factor) * grid_position_of_A -
-        rotation_error_cross_grid_pos_A);
+    const tnsr::I<double, 3, Frame::Grid>& grid_position_of_B_tnsr =
+        Parallel::get<domain::Tags::ObjectCenter<domain::ObjectLabel::B>>(
+            cache);
+    const DataVector grid_position_of_B{{grid_position_of_B_tnsr[0],
+                                         grid_position_of_B_tnsr[1],
+                                         grid_position_of_B_tnsr[2]}};
+    const DataVector& current_position_of_B = get<center_B>(measurements);
+
+    const DataVector grid_position_average =
+        0.5 * (grid_position_of_A + grid_position_of_B);
+    const DataVector current_position_average =
+        0.5 * (current_position_of_A + current_position_of_B);
+
+    const DataVector grid_separation = grid_position_of_A - grid_position_of_B;
+    const DataVector current_separation =
+        current_position_of_A - current_position_of_B;
+
+    // These quantities come from the translation control implementation in SpEC
+    double current_separation_dot_grid_separation =
+        dot(current_separation, grid_separation);
+    double current_separation_dot_grid_average =
+        dot(current_separation, grid_position_average);
+    double grid_separation_dot_grid_average =
+        dot(grid_separation, grid_position_average);
+    double grid_separation_dot_grid_separation =
+        dot(grid_separation, grid_separation);
+
+    // From eq. 42 in 1304.3067 where the grid and current position are swapped
+    // from only object A to the average grid and current positions of both
+    // objects.
+    DataVector translation_control =
+        make_with_value<DataVector>(grid_position_of_A, 0.0);
+    translation_control =
```

Comment:
[nitpick] No need to allocate first, then assign. Just do it all in one step

Reply:
oh whoops yeah, forgot to take that out with the loop